public/app/features/plugins/admin/components/PluginDetailsBody.tsx

```diff
@@ -1,65 +1,68 @@
 import React from 'react';
 import { css, cx } from '@emotion/css';
 
-import { AppPlugin, GrafanaTheme2, GrafanaPlugin, PluginMeta } from '@grafana/data';
+import { AppPlugin, GrafanaTheme2 } from '@grafana/data';
 import { useStyles2 } from '@grafana/ui';
 
-import { PluginTabLabels } from '../types';
+import { CatalogPlugin, PluginTabLabels } from '../types';
 import { VersionList } from '../components/VersionList';
+import { usePluginConfig } from '../hooks/usePluginConfig';
 import { AppConfigCtrlWrapper } from '../../wrappers/AppConfigWrapper';
 import { PluginDashboards } from '../../PluginDashboards';
 
-type PluginDetailsBodyProps = {
+type Props = {
   tab: { label: string };
-  plugin: GrafanaPlugin<PluginMeta<{}>> | undefined;
-  remoteVersions: Array<{ version: string; createdAt: string }>;
-  readme: string;
+  plugin: CatalogPlugin;
 };
 
-export function PluginDetailsBody({ tab, plugin, remoteVersions, readme }: PluginDetailsBodyProps): JSX.Element | null {
+export function PluginDetailsBody({ tab, plugin }: Props): JSX.Element | null {
   const styles = useStyles2(getStyles);
+  const { value: pluginConfig } = usePluginConfig(plugin);
 
   if (tab?.label === PluginTabLabels.OVERVIEW) {
     return (
       <div
         className={cx(styles.readme, styles.container)}
-        dangerouslySetInnerHTML={{ __html: readme ?? 'No plugin help or readme markdown file was found' }}
+        dangerouslySetInnerHTML={{
+          __html: plugin.details?.readme ?? 'No plugin help or readme markdown file was found',
+        }}
       />
     );
   }
 
   if (tab?.label === PluginTabLabels.VERSIONS) {
     return (
       <div className={styles.container}>
-        <VersionList versions={remoteVersions ?? []} />
+        <VersionList versions={plugin.details?.versions} />
       </div>
     );
   }
 
-  if (tab?.label === PluginTabLabels.CONFIG && plugin?.angularConfigCtrl) {
+  if (tab?.label === PluginTabLabels.CONFIG && pluginConfig?.angularConfigCtrl) {
     return (
       <div className={styles.container}>
-        <AppConfigCtrlWrapper app={plugin as AppPlugin} />
+        <AppConfigCtrlWrapper app={pluginConfig as AppPlugin} />
       </div>
     );
   }
 
-  if (plugin?.configPages) {
-    for (const configPage of plugin.configPages) {
+  if (pluginConfig?.configPages) {
+    for (const configPage of pluginConfig.configPages) {
       if (tab?.label === configPage.title) {
         return (
           <div className={styles.container}>
-            <configPage.body plugin={plugin} query={{}} />
+            {/* TODO: we should pass the query params down */}
```

Comment:
Something we should do within this PR or in a follow up PR?

Reply:
I would personally fix this in an upcoming small PR if that's not a problem.

(Mostly just to not make this PR bigger and also because it is not something that was introduced now, but something that caught our eye and wanted to put a reminder on it.)

Is it ok if I create an issue for it?

Reply:
Yes, no worries! I just asked to make sure we didn't forgot about it :)